staging/src/k8s.io/apiserver/pkg/admission/configuration/validating_webhook_manager.go

```diff
@@ -75,23 +96,40 @@ func (v *validatingWebhookConfigurationManager) getConfiguration() ([]webhook.We
 	if err != nil {
 		return []webhook.WebhookAccessor{}, err
 	}
-	return mergeValidatingWebhookConfigurations(configurations), nil
+	return v.smartReloadValidatingWebhookConfigurations(configurations), nil
 }
 
-func mergeValidatingWebhookConfigurations(configurations []*v1.ValidatingWebhookConfiguration) []webhook.WebhookAccessor {
+func (v *validatingWebhookConfigurationManager) smartReloadValidatingWebhookConfigurations(configurations []*v1.ValidatingWebhookConfiguration) []webhook.WebhookAccessor {
 	sort.SliceStable(configurations, ValidatingWebhookConfigurationSorter(configurations).ByName)
 	accessors := []webhook.WebhookAccessor{}
 	for _, c := range configurations {
+		cachedConfigurationAccessors, ok := v.configurationsCache.Load(c.Name)
+		if ok {
+			// Pick an already cached webhookAccessor
+			accessors = append(accessors, cachedConfigurationAccessors.([]webhook.WebhookAccessor)...)
+			continue
+		}
+
 		// webhook names are not validated for uniqueness, so we check for duplicates and
 		// add a int suffix to distinguish between them
+		//
+		// NOTE: In `pkg/apis/admissionregistration/validation.go` webhook names are checked
+		// for uniqueness now. Is it safe to remove this?
+		// If we ever get rid of this, we can change the cache map value type to
+		// map[string]WebhookAccessor instead. Where keys will be of format "config.name/webhook.name"
```

Comment:
I don't think we'd change this to a map and lose ordering, can probably drop this comment